mergify_engine/actions/utils.py

```diff
@@ -53,12 +54,20 @@ async def validate_bot_account(
 
     if required_permissions:
         # TODO(sileht): Cache this, people only use one bot account!
-        permission = typing.cast(
-            github_types.GitHubRepositoryCollaboratorPermission,
-            await ctxt.client.item(
-                f"{ctxt.base_url}/collaborators/{bot_account}/permission"
-            ),
-        )["permission"]
+        try:
+            permission = typing.cast(
+                github_types.GitHubRepositoryCollaboratorPermission,
+                await ctxt.client.item(
+                    f"{ctxt.base_url}/collaborators/{bot_account}/permission"
+                ),
+            )["permission"]
+        except http.HTTPNotFound:
```

Comment:
This looks good though unrelated no?

Reply:
Yes and no.
No because before the rebase action didn't check for the permission and just fail on git push, now I check the permission upfront, so I had to handle the error here, since a test was covering this case.
And yes because, the merge action already uses this code and may have a bug if the user doesn't here.